### PR TITLE
docs: Add a test for POST with TurboStreams

### LINKED ISSUE #681: Hotwired Turbo broken in Micronaut 4.x (works in 3.10.3)
### Expected Behavior

A controller action that consumes MULTIPART_FORM_DATA and produces TEXT_HTML using TurboFrameView should work like it used to do in Micronaut 3.10.3

### Actual Behaviour

In Micronaut 4.0.0 and 4.2.3 this simple controller action
```java
@Consumes(MediaType.MULTIPART_FORM_DATA)
@TurboFrameView("view")
@Produces(value = MediaType.TEXT_HTML)
@Post("/edit")
Map<String, Object> processEdit(@Part int id, @Part String title, @Part String body) {
        var episode = Episode.builder().id(id).title(title).body(body).build();
    repository.update(episode);
    return Map.of(
            "episode", episode,
            "episodes", repository.loadAllEpisodes());
}
```

results in "No possible encoders found for media type: text/html"
```
io.micronaut.http.codec.CodecException: Cannot encode value [{episodes=[Episode(id=1, title=Jamming With Edward!), Episode(id=2, title=Bohemian Rhapsody)], episode=Episode(id=1, title=Jamming With Edward!)}]. No possible encoders found for media type: text/html
	at io.micronaut.http.body.DynamicMessageBodyWriter.find(DynamicMessageBodyWriter.java:69)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$CompatNettyWriteClosure.writeTo(RoutingInBoundHandler.java:519)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.writeNettyMessageBody(RoutingInBoundHandler.java:359)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.encodeHttpResponse(RoutingInBoundHandler.java:338)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.writeResponse(RoutingInBoundHandler.java:233)
	at io.micronaut.http.server.netty.NettyRequestLifecycle.lambda$handleNormal$0(NettyRequestLifecycle.java:87)
	at io.micronaut.http.reactive.execution.ReactorExecutionFlowImpl$1.onComplete(ReactorExecutionFlowImpl.java:121)
	at reactor.core.publisher.MonoFlatMap$FlatMapMain.onNext(MonoFlatMap.java:159)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.FluxOnErrorResume$ResumeSubscriber.onNext(FluxOnErrorResume.java:79)
	at reactor.core.publisher.FluxMap$MapSubscriber.onNext(FluxMap.java:122)
	at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:82)
	at reactor.core.publisher.MonoNext$NextSubscriber.onNext(MonoNext.java:82)
	at reactor.core.publisher.Operators$MonoInnerProducerBase.complete(Operators.java:2781)
	at reactor.core.publisher.SinkOneMulticast.subscribe(SinkOneMulticast.java:99)
	at reactor.core.publisher.MonoDeferContextual.subscribe(MonoDeferContextual.java:55)
	at io.micronaut.core.async.propagation.ReactivePropagation$1.subscribe(ReactivePropagation.java:54)
	at io.micronaut.core.async.propagation.ReactivePropagation$1.subscribe(ReactivePropagation.java:61)
	at reactor.core.publisher.MonoFromPublisher.subscribe(MonoFromPublisher.java:67)
	at reactor.core.publisher.InternalMonoOperator.subscribe(InternalMonoOperator.java:64)
	at io.micronaut.http.reactive.execution.ReactorExecutionFlowImpl.onComplete(ReactorExecutionFlowImpl.java:89)
	at io.micronaut.http.server.netty.NettyRequestLifecycle.handleNormal(NettyRequestLifecycle.java:87)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.accept(RoutingInBoundHandler.java:220)
	at io.micronaut.http.server.netty.handler.PipeliningServerHandler$OptimisticBufferingInboundHandler.read(PipeliningServerHandler.java:422)
	at io.micronaut.http.server.netty.handler.PipeliningServerHandler.channelRead(PipeliningServerHandler.java:206)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
	at io.netty.handler.codec.MessageToMessageCodec.channelRead(MessageToMessageCodec.java:111)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.ChannelInboundHandlerAdapter.channelRead(ChannelInboundHandlerAdapter.java:93)
	at io.netty.handler.codec.http.HttpServerKeepAliveHandler.channelRead(HttpServerKeepAliveHandler.java:64)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelRead(CombinedChannelDuplexHandler.java:436)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318)
	at io.netty.channel.CombinedChannelDuplexHandler.channelRead(CombinedChannelDuplexHandler.java:251)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

### Steps To Reproduce

1. build and run the [demo application](https://github.com/PeterFokkinga/micronaut-hotwired-bug), default it uses Micronaut 4.2.3
2. click on a link in the "Episodes" frame, notice how only the "One Episode" frame refreshes (as expected)
3. click on the "edit" link, notice how only this frame refreshes, now showing an edit form
4. change something and click the "save" button: nothing happens but see the logs for exception

**now, change the Micronaut version to 3.10.3 and the plugin version to 3.7.10 and rebuild with --refresh-dependencies**

Repeat the steps above, but notice how step 4 now works as expected: it changes the contents of the "One Episode" to the view of the data you just edited.

### Environment Information

_No response_

### Example Application

https://github.com/PeterFokkinga/micronaut-hotwired-bug

### Version

4.2.3

### PR DESCRIPTION
**Pull request by timyates**
This also moves common classes out of the tests themselves. And sorts out the layout in the tests, by adding indent=0 to the snippet tags.

Closes #681

### CODE REVIEW

File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/turbo/TurboFrameTest.kt [hunk 1]
```diff
@@ -34,33 +44,58 @@ class TurboFrameTest {
...
+            HttpRequest.POST("/frame/messages/1", body)
+                .contentType(MediaType.MULTIPART_FORM_DATA)
+                .accept(TurboMediaType.TURBO_STREAM),
+            String::class.java
+        )
+        assertEquals(
+            "<turbo-stream action=\"update\"><template><h1>My new title</h1>\n" +
+                    "<p>My new body</p>\n" +
+                    "</template></turbo-stream>", html
+        )
```
Comment (timyates) [inline test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/turbo/TurboFrameTest.kt:74]: This is the test

File: test-suite/src/test/java/io/micronaut/views/docs/turbo/TurboFrameTest.java [hunk 2]
```diff
@@ -59,66 +67,64 @@ void youCanUseTurboFrameToRenderFrame() {
...
+            .addPart("body", "My new body")
+            .build();
+        html = client.retrieve(
+            HttpRequest.POST("/frame/messages/1", body)
+                .contentType(MediaType.MULTIPART_FORM_DATA)
+                .accept(TurboMediaType.TURBO_STREAM),
+            String.class);
+        assertEquals("<turbo-stream action=\"update\"><template><h1>My new title</h1>\n" +
+            "<p>My new body</p>\n" +
+            "</template></turbo-stream>", html);
```
Comment (timyates) [inline test-suite/src/test/java/io/micronaut/views/docs/turbo/TurboFrameTest.java:88]: This is the test

File: test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/turbo/TurboFrameTest.kt [hunk 3]
```diff
@@ -34,33 +44,58 @@ class TurboFrameTest {
...
+        //tag::turboFramePost[]
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        @TurboView("view")
+        @Produces(value = [MediaType.TEXT_HTML, TurboMediaType.TURBO_STREAM])
+        @Post("/messages/{id}")
+        fun processEdit(@Part id: Int, @Part title: String, @Part body: String): Map<String, Any> {
+            // Process the posted data, and return the updated message
+            return mapOf("message" to Message(id.toLong(), title, body))
+        }
+        //end::turboFramePost[]
```
Comment (timyates) [inline test-suite-kotlin/src/test/kotlin/io/micronaut/views/docs/turbo/TurboFrameTest.kt:99]: And this is the post handler in in the test and the docs

File: test-suite/src/test/java/io/micronaut/views/docs/turbo/TurboFrameTest.java [hunk 4]
```diff
@@ -59,66 +67,64 @@ void youCanUseTurboFrameToRenderFrame() {
...
+        //tag::turboFramePost[]
+        @Consumes(MediaType.MULTIPART_FORM_DATA)
+        @TurboView("view")
+        @Produces(value = {MediaType.TEXT_HTML, TurboMediaType.TURBO_STREAM})
+        @Post("/messages/{id}")
+        Map<String, Object> processEdit(@Part int id, @Part String title, @Part String body) {
+            // Process the posted data, and return the updated message
+            return Map.of("message", new Message((long) id, title, body));
         }
+        //end::turboFramePost[]
```
Comment (timyates) [inline test-suite/src/test/java/io/micronaut/views/docs/turbo/TurboFrameTest.java:114]: And this is the post handler in in the test and the docs

**Review by sdelamo — APPROVED** [review]

**Pull request merged**